doc/examples/segmentation/plot_euler_number.py

```diff
@@ -0,0 +1,147 @@
+"""
+=========================
+Euler number
+=========================
+
+This example shows an illustration of the computation of the Euler number [1]_
+in 2D and 3D objects.
+
+For 2D objects, the Euler number is the number of objects minus the number of
+holes. Notice that if an object is 8-connected, its complementary set is
+4-connected, and conversely. It is also possible to compute the number of
+objects using :func:`skimage.measure.label`, and to deduce the number of holes
+from the difference between the two numbers.
```

Comment:
Just to clarify my understanding here? "complementary set" refers to everything that is not part of the object? I'm not to happy about an "object being x-connected". The neighborhood of each pixel of the object is 8-connected, not the object in itself, which was confusing me initially. :sweat_smile: 

How about:

"Notice that if an object is formed by a neighborhood of 8 connected pixels (2-connectivity), its complementary set (holes, background) are formed by a neighborhood of 4 connected pixels (1-connectivity), and conversely."

It's semantics and maybe I'm just confused because of limited terminology, so take it as an optional suggestion. :wink:

Reply:
I like this proposal since we want to promote the use of 1, 2, 3-connectivity rather than neighbourhood counting.

Reply:
the PR has been merged but I'll do the modification in a follow-up PR

Reply:
please see #4989